### DISCUSSION TITLE
Slides flash one under the other before settling to show them next to each other

### DISCUSSION THREAD

**Discussion by kevinrenskers** [nolimits4web/swiper#6828] [Q&A]
When the carousel initially loads on screen, you see all slides one under each other for a brief moment, before they quickly settle to be shown one next to the other.

I'm seeing the same behavior in Safari, Chrome and Firefox. It's easily visible with a very minimal example:

https://github.com/nolimits4web/swiper/assets/229384/c86066f7-f0b3-487a-a230-8cad6650ef09

To run the example:

1. Checkout https://github.com/kevinrenskers/swiper-repro-2
2. npm i
3. npm run dev

But in a real-world website it's much worse:

https://github.com/nolimits4web/swiper/assets/229384/04d23c62-a94d-4ec9-af37-ec783052654a

In my opinion this seems like a bug. The slider should be shown with all the slides already in their correct position, but instead you get this "flash of unpositioned content". However my bug report was promptly closed without comment other than pointing to the discussions and Stackoverflow, which is disappointing. How is this not a bug?

**Comment by nolimits4web** [comment]
This means you initialise it too late, init it as soon as the DOM is ready, and there will no be any flickers, or hide it on your own and show when it is ready

**Reply by kevinrenskers** [reply]
I'm just using normal Svelte / SvelteKit components, see GitHub repo. Not sure how I would init it sooner?

**Reply by kevinrenskers** [reply]
In fact I am using the exact same methods as documented here: https://swiperjs.com/element#usage-with-svelte.

**Reply by TomasCarlson** [reply]
This is an issue we're seeing with Sveltekit too. A workaround to make it not as apparent is to set the swiper-container to flex, but that's not ideal

**Reply by pavelstianko** [reply]
Yes, we are struggling with the same problem on our websites in Sveltekit. I tried almost all ideas how to solve this, but nothing is working. My only solution now is to set width in css for each slide (that is changed afted init with real inline width).

**Comment by zygo1** [comment]
It might be late, but a solution about this problem is this one below: 

```html
<!--  Initially hidden -->
<div class="swiper w-full h-full opacity-0" ref={ref}>
  <div class="swiper-wrapper">
    {slides.map(...)}
  </div>
</div>


We added opacity-0 in the parent div, to make it hidden initially

And then there is an another property in the docs of swiper.js which let's you handle the initialization, so you remove the opacity-0 on init and your slider will be rendered when it is fully initialized. 


const element = ref.current;
const swiper = new Swiper(element, {
  modules: [Navigation, Autoplay],
  slidesPerView: 1,
  loop: true,
  // ...other options...
  on: {
    init: () => {
      // Reveal the slider once it's fully initialized
      element.classList.remove("opacity-0");
    }
  }
});
